extensions/awork/src/composables/WebClient.ts

```diff
@@ -0,0 +1,168 @@
+import { Alert, confirmAlert, getPreferenceValues, LocalStorage, OAuth, PreferenceValues } from "@raycast/api";
+import { showFailureToast } from "@raycast/utils";
+import fetch, { RequestInit } from "node-fetch";
+
+interface workspace {
+  id: string;
+  name: string;
+  url: string;
+}
+
+interface User {
+  id: string;
+  workspace: workspace;
+}
+
+export const baseURI = "https://api.awork.com/api/v1";
+export let authorizationInProgress = false;
+export let revalidating = false;
+
+const preferences = getPreferenceValues<PreferenceValues>();
```

Comment:
`Preferences` is auto-generated and is type-safe.
```suggestion
const preferences = getPreferenceValues<Preferences>();
```